tests/test_wikidata_gazetteer.py

```
@@ -0,0 +1,276 @@
+
+
+def test_lookup_rejects_backslash_in_name(wd_gaz):
+    with pytest.raises(ValueError, match="illegal character"):
+        wd_gaz.lookup("Foo\\bar")
+
+
+def test_lookup_no_match_raises_not_found(wd_gaz):
+    from siege_utilities.geo.gazetteers.base import GazetteerNotFoundError
+
+    with patch.object(wd_gaz._session, "get", return_value=_sparql_response([])):
+        with pytest.raises(GazetteerNotFoundError, match="no match"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)